packages/data-context/src/actions/WizardActions.ts

```diff
@@ -28,28 +29,33 @@ export class WizardActions {
     return this.ctx.wizardData
   }
 
-  setFramework (framework: NexusGenEnums['FrontendFrameworkEnum'] | null) {
-    const prevFramework = this.ctx.coreData.wizard.chosenFramework || ''
+  setFramework (framework: typeof FRONTEND_FRAMEWORKS[number]['type'] | null): void {
+    const next = FRONTEND_FRAMEWORKS.find((x) => x.type === framework)
 
     this.ctx.coreData.wizard.chosenFramework = framework
 
-    if (framework !== 'react' && framework !== 'vue') {
-      return this.setBundler('webpack')
+    if (next?.supportedBundlers?.length === 1) {
+      this.setBundler(next?.supportedBundlers?.[0].type)
+
+      return
     }
 
     const { chosenBundler } = this.ctx.coreData.wizard
 
     // if the previous bundler was incompatible with the
-    // new framework, we need to reset it
-    if ((chosenBundler && !this.ctx.wizard.chosenFramework?.supportedBundlers.includes(chosenBundler))
-    || !['react', 'vue'].includes(prevFramework)) {
-      return this.setBundler(null)
-    }
+    // new framework that was selected, we need to reset it
+    const doesNotSupportChosenBundler = (chosenBundler && !new Set(
+      this.ctx.wizard.chosenFramework?.supportedBundlers.map((x) => x.type) || [],
+    ).has(chosenBundler)) ?? false
+
+    const prevFramework = this.ctx.coreData.wizard.chosenFramework || ''
 
-    return
+    if (doesNotSupportChosenBundler || !['react', 'vue'].includes(prevFramework)) {
+      this.setBundler(null)
+    }
   }
 
-  setBundler (bundler: NexusGenEnums['SupportedBundlers'] | null) {
```

Comment:
I don't think we should use nexus codegen types in `data-context`... rather we should use the ones we define ourselves. I would like to lint for this in the future.